aiohttp_client_cache/backends/sqlite.py

```diff
@@ -120,22 +112,27 @@ async def bulk_commit(self):
             ...         await cache.write(f'key_{i}', str(i))
 
         """
-        self._bulk_commit = True
-        self._connection = await aiosqlite.connect(self.filename, **self.connection_kwargs)
+        async with self._lock:
+            self._bulk_commit = True
         try:
             yield
             await self._connection.commit()
         finally:
-            self._bulk_commit = False
-            await self._connection.close()
-            self._connection = None
+            async with self._lock:
+                self._bulk_commit = False
```

Comment:
I don't think the locks help here if it's to avoid changing the value of `self._bulk_commit` while a bulk commit is in progress. Imagine there's a short- and a long-running bulk commit, both set `self._bulk_commit` to true on entry, and the one that finishes first flips it back to false.  All of the long-running `bulk_commit`'s operations will now commit upon releasing `get_connection`.

Reply:
The way I would approach this would be by using a [context var](https://docs.python.org/3/library/contextvars.html) instead of an instance attribute.

Reply:
Thanks! That's very helpful.

Reply:
Fixed, and added a test for that.

Reply:
LGTM :)

I don't know if there are any downsides to creating a context var on object initialisation - the Python docs recommend having them at the "top module level".

Reply:
Done.